jupyterlab_git/git.py

```diff
@@ -435,29 +482,59 @@ def checkout_all(self, top_repo_path):
         )
         return my_output
 
-    def commit(self, commit_msg, top_repo_path):
+    def commit(self, commit_msg, top_repo_path, author_name=None, author_email=None):
         """
         Execute git commit <filename> command & return the result.
         """
-        my_output = subprocess.check_output(
-            ["git", "commit", "-m", commit_msg], cwd=top_repo_path
+        command = ["git", "commit", "-m", commit_msg]
+
+        if author_name != None and author_email != None:
```

Comment:
How is this related to the auth changes? Is it necessary for them?

Reply:
This is related to auth, since user should be able to provide commit information without going to Terminal. This is not necessary, but without it the user flow breaks when they click commit button.

Reply:
It is important that Git commits get attributed to a specific author, particularly in multi-user repositories. How will users set the `name`  and `email` setting for this attribution?

Quick examples:
- https://help.github.com/en/articles/setting-your-username-in-git
- https://help.github.com/en/articles/setting-your-commit-email-address-in-git

Reply:
@brylie In the proposed implementation each commit is attributed individually in favor of global setting:
```
git commit -c user.name <name> -c user.email <email>
```

Global option could be added as described in the links you provided. It could a check box in the interface ("Save globally"). Let me know if you think I should enable global setting or the local one is fine for now

Reply:
I think a global settings makes sense to have a global flag here that is seperate from this auth support. Maybe an item in the git menu that lets you edit and set your username an email in the config.

But I am still unclear, can't we include the auth support without this extract logic around the name and email? And just assume people have already set that up globally?

Reply:
I took a look at the code. I think this looks fine as is. We can always move it to setting the global later.